### PR TITLE
Btsc 375/google account auth option

### PR DESCRIPTION
**Pull request by AlexMCon**

### CODE REVIEW

File: src/types/generic.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum PlatformTypes {
```
Comment (obasilakis) [inline src/types/generic.ts:1]: Is it possible to rename this file to `shared` or `common`?

File: src/controllers/provider.controller.ts [hunk 2]
```diff
@@ -52,7 +57,7 @@ export const provider_auth = async (request: Request, response: Response) => {
 
   if (getAddressHash(address.toLowerCase()) !== provider.walletAddressHashed) {
     return response
-      .status(401)
+      .status(400)
```
[the remark this answers is not in the document]
Comment (AlexMCon) [reply src/controllers/provider.controller.ts:60]: We got 2 places where we have this - one is when the users logs in and one is when he links the wallet. For the first instance it should be 401, unauthorized, for the second instance 400 is better from 2 perspectives - one is that you aren't auto logged out on the front-end (when you receive 401 you are logged out directly) and 2 we differentiate better between the cases like this on both back and front so we can debug easier.

TL;DR changed to 401 where the user logs in and let 400 where he links on both provider and assessor.

File: src/types/generic.ts (hunk 1 above)
Comment (AlexMCon) [reply src/types/generic.ts]: Changed everything else according to your suggestions

**Review by obasilakis — APPROVED** [review]
